packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -164,29 +172,44 @@ export interface ITreeCheckout extends AnchorLocator {
  * @remarks This does not create a {@link SharedTree}, but rather a view with the minimal state
  * and functionality required to implement {@link ITreeCheckout}.
  */
-export function createTreeCheckout(args?: {
-	branch?: SharedTreeBranch<SharedTreeEditBuilder, SharedTreeChange>;
-	changeFamily?: ChangeFamily<SharedTreeEditBuilder, SharedTreeChange>;
-	schema?: TreeStoredSchemaRepository;
-	forest?: IEditableForest;
-	events?: ISubscribable<CheckoutEvents> &
-		IEmitter<CheckoutEvents> &
-		HasListeners<CheckoutEvents>;
-	removedRoots?: DetachedFieldIndex;
-}): TreeCheckout {
+export function createTreeCheckout(
+	idCompressor: IIdCompressor,
+	args?: {
+		branch?: SharedTreeBranch<SharedTreeEditBuilder, SharedTreeChange>;
+		changeFamily?: ChangeFamily<SharedTreeEditBuilder, SharedTreeChange>;
+		schema?: TreeStoredSchemaRepository;
+		forest?: IEditableForest;
+		fieldBatchCodec?: FieldBatchCodec;
+		events?: ISubscribable<CheckoutEvents> &
+			IEmitter<CheckoutEvents> &
+			HasListeners<CheckoutEvents>;
+		removedRoots?: DetachedFieldIndex;
+	},
+): TreeCheckout {
 	const forest = args?.forest ?? buildForest();
+	const schema = args?.schema ?? new TreeStoredSchemaRepository();
+	const defaultCodecOptions = { jsonValidator: noopValidator };
 	const changeFamily =
-		args?.changeFamily ?? new SharedTreeChangeFamily({ jsonValidator: noopValidator });
+		args?.changeFamily ??
+		new SharedTreeChangeFamily(
+			idCompressor,
+			args?.fieldBatchCodec ??
+				makeFieldBatchCodec(defaultCodecOptions, {
+					// TODO: provide schema here to enable schema based compression.
+					encodeType: TreeCompressionStrategy.Compressed,
+				}),
+			{ jsonValidator: noopValidator },
+		);
 	const branch =
 		args?.branch ??
 		new SharedTreeBranch(
 			{
 				change: changeFamily.rebaser.compose([]),
-				revision: assertIsRevisionTag("00000000-0000-4000-8000-000000000000"),
+				revision: idCompressor.generateCompressedId(),
```

Comment:
Should this be `"root"`?

Reply:
yes. turns out it doesn't matter based on usage (hence why no test failure). but "root" is definitely what should be here.